### PR TITLE
Feature/readme

### PR DESCRIPTION
**Pull request by befirst**
You can look at README in formatted view here: https://github.com/Dash-Kit/dash-kit-core/tree/feature/readme

### CODE REVIEW

**Review by w3ggy — APPROVED** [review]

**Review by MrtnvM — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -4,6 +4,289 @@ The library providing basic architecture components.
...
-[async_redux](https://pub.dev/packages/async_redux)
+This is how `async_redux` flow looks like:
+
+![Flow Diagram](https://miro.medium.com/max/1400/1*O34Rwgxfa3NiE7F_QW683Q.png)
+
+So the conceptual difference with classical redux here is:
+
+#### Async Reducer
+
+If you want to do some asynchronous work, you simply declare the action’reducer to return Future<AppState> then change the state and return it. There is no need of any "middleware", like for other Redux versions.
```
Comment (w3ggy) [inline README.md:17]: `the action’reducer`

File: README.md [hunk 2]
```diff
@@ -4,6 +4,289 @@ The library providing basic architecture components.
...
+- void addAll(Iterable<T> values)
+- void updateItem(Object id, T value)
+- void deleteItem(Object id)
+- void clear()
+
+## PaginatedList
+
+To incapsulate data for lists that you can download page by page, we wrote `PaginatedList`.
+
+Ussualy you can meet the next meta information for paginated resources:
```
Comment (w3ggy) [inline README.md:244]: Usually,

File: README.md [hunk 3]
```diff
@@ -4,6 +4,289 @@ The library providing basic architecture components.
...
+Future<AppState> reduce() async {
+  return state.dataStructure
+          .someOldList
+          // someOldList will be rebuilt
+          .deleteItem(conditionItemId);
+}
+```
+
+### Availabe API
+All of this methods will rebuild the main list, so your list will be immutable, and you don't need to call `rebuild`:
```
Comment (w3ggy) [inline README.md:231]: these

File: README.md [hunk 4]
```diff
@@ -4,6 +4,289 @@ The library providing basic architecture components.
...
+  void onLogin(String email, String password) {
+    dispatch(action)
+      // We can route to some page here
+      .then((value) => _onLoginSuccess())
+      // And show some dialog here
+      .catchError((error) => _onLoginError();
+  }
+}
+```
+`LoadableView` is very simple widget which displays progress indicator above main content, you can find that widget on [GitHub here](https://github.com/Dash-Kit/dash-kit-loadable) or on [pub.dev here](https://pub.dev/packages/dash_kit_loadable).
```
Comment (w3ggy) [inline README.md:180]: is a

**Pull request merged**